### PR TITLE
More performance improvements

### PR DESCRIPTION
**Pull request by lindsay-stevens**
A few more general ideas following #740

#### Why is this the best possible solution? Were any other approaches considered?

See commit messages for details of each change. Overall a modest but worthwhile improvement. I think to get much more will require addressing inefficiencies in XML generation, which may require overriding more of of minidom's Element class and perhaps Attr as well. Also the relative repeats recursion issue remains.

#### What are the regression risks?

Should be minimal risk, the only tests changes were to update the performance test stats.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.

No

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: pyxform/question.py [hunk 1]
```diff
@@ -391,6 +391,22 @@ def validate(self):
             for child in self.children:
                 child.validate()
 
+    def iter_descendants(
```
Comment (lognaturel) [inline pyxform/question.py:388]: Maybe this implementation could go in some sort of shared place? Everyone's favorite, a util file? I'm worried about these identical implementations drifting apart...

**Review by lognaturel — APPROVED** [review]
Nice! 🏃‍♀️

File: pyxform/question.py (hunk 1 above)
Comment (lindsay-stevens) [reply pyxform/question.py:388]: Agree 3x copy/pastas = probably time for a shared func. I didn't do that here because of the points noted in #745. In which case the only copy of this method override for `iter_descendants` would be MultipleChoice.
